lab_1/main.py

```diff
@@ -104,10 +154,32 @@ def compare_profiles_advanced(unknown_profile: dict, profile_to_compare: dict, t
     :return: a dictionary with 7 keys – name, score, common, sorted_common, max_length_word,
     min_length_word, average_token_length
     """
-    pass
-
-
-def detect_language_advanced(unknown_profile: dict, profiles: list, languages: list, top_n: int) -> str or None:
+    if not (isinstance(unknown_profile, dict)
+            and isinstance(profile_to_compare, dict)
+            and isinstance(top_n, int)):
+        return None
+    top_n_words_unknown_profile = get_top_n_words(unknown_profile['freq'], top_n)
+    top_n_words_profile_to_compare = get_top_n_words(profile_to_compare['freq'], top_n)
+    common = []
+    for i in top_n_words_profile_to_compare:
+        if i in top_n_words_unknown_profile:
+            common.append(i)
+    score = float(len(common) / len(top_n_words_unknown_profile))
+    sorted_common = sorted(common)
+    sorted_tokens = sorted(list(profile_to_compare['freq'].keys()), key=len)
```

Comment:
can you explain what happens here in writing?

Reply:
In line 168 we sort the list of common words in alphabetical order.
In line 169 we take the tokens of known language from its profile (more accurate to say, we take them from its frequency dictionary, so we need keys) and then, we sort this tokens by length to find words with minimum and maximum length.